merlin/models/tf/models/base.py

```diff
@@ -311,15 +311,7 @@ def compile(
         if num_v1_blocks > 0:
             self.output_names = [task.task_name for task in self.prediction_tasks]
         else:
-            self.output_names = [block.full_name for block in self.prediction_blocks]
-            negative_sampling = kwargs.pop("negative_sampling", None)
-            if negative_sampling:
-                if not isinstance(self.prediction_blocks[0], ContrastivePredictionBlock):
-                    raise ValueError(
-                        "Negative sampling strategy can be used only with a"
-                        " `ContrastivePredictionBlock` prediction block"
-                    )
-                self.prediction_blocks[0].compile(negative_sampling=negative_sampling)
```

Comment:
was this explicit compile unnecessary before?

Reply:
It was something we had before as well, but it wasn’t used anywhere in the tests.